eclipse-scout-core/test/lookup/RestLookupCallSpec.js

```diff
@@ -0,0 +1,327 @@
+    expect(cloneText._getRestrictionForAjaxCall()).toEqual({
+      ...expectedDefaultRestriction,
+      text: 'xyz',
+      myProperty: 'someValue'
+    });
+
+    const cloneRec = lookupCall.cloneForRec(789);
+    expect(cloneRec.restriction).toBeNull();
+    expect(cloneRec._restriction).toEqual(expectedDefaultRestriction);
+    expect(cloneRec.parentKey).toBe(789); // not (yet) on the restriction!
+    cloneRec.addRestriction('myProperty', 'someValue');
+    // TODO uncomment the lines marked with '<--' when getByRec() is implemented for RestLookupCall.js!
+    // cloneRec.execute(); <-- uncomment
+    expect(cloneRec._getRestrictionForAjaxCall()).toEqual({
+      ...expectedDefaultRestriction,
+      // parentKey: 789, <-- uncomment
+      myProperty: 'someValue'
+    });
+
+    const cloneKey = lookupCall.cloneForKey(123);
+    expect(cloneKey.restriction).toBeNull();
+    expect(cloneKey._restriction).toBeNull(); // different from all, text and rec
+    expect(cloneKey.key).toBe(123); // not (yet) on the restriction!
+    cloneKey.execute();
+    cloneKey.addRestriction('myProperty', 'someValue');
+    expect(cloneKey._getRestrictionForAjaxCall()).toEqual({
+      ids: [123],
+      myProperty: 'someValue'
+    });
+
+    const cloneKeys = lookupCall.cloneForKeys([123, 456]);
+    expect(cloneKeys.restriction).toBeNull();
+    expect(cloneKeys._restriction).toBeNull(); // different from all, text and rec
+    expect(cloneKeys.keys).toEqual([123, 456]); // not (yet) on the restriction!
+    cloneKeys.addRestriction('myProperty', 'someValue');
+    cloneKeys.execute();
+    expect(cloneKeys._getRestrictionForAjaxCall()).toEqual({
+      ids: [123, 456],
+      myProperty: 'someValue'
+    });
+
+    // Check template again (must not have been altered)
+    expect(lookupCall.restriction).toBeNull();
+    expect(lookupCall._restriction).toBeNull();
+  });
+
+  it('applies restrictions in the expected order', () => {
+    // This test asserts that the following precedence rules are applies then creating the final restriction object:
+    // 1. lookup call model (for maxRowCount)
+    // 2. predefined restriction
+    // 3. dynamic restrictions added by cloneForX()
+    // 4. dynamic restrictions added by user code (e.g. during 'prepareLookupCall' event)
+
+    const lookupCall = scout.create('RestLookupCall', {
+      session: session,
+      resourceUrl: 'test-api/dummy',
+      maxRowCount: 777,
+      active: true, // will be ignored
+      myProperty: 'one',
+      myIgnoredProperty: 'three', // will be ignored
+      restriction: {
+        maxRowCount: 888,
+        active: false,
+        myProperty: 'two',
+        myDefaultProperty: 'three'
+      }
+    });
+    spyOn(lookupCall, '_call'); // <-- disable the ajax call in this test
+
+    const expectedModelRestriction = {
+      maxRowCount: 888,
+      active: false,
+      myProperty: 'two',
+      myDefaultProperty: 'three'
+    };
+
+    const expectedDefaultRestriction = {
```

Comment:
looks a bit strange to have two semantic identical objects => Add a comment to describe the position change in the ajax call restriction

Reply:
Semantically, they are not exactly the same, but in this particular case, they contain exactly the same properties (the order is not important). There is no default property left that could be used to show that difference, therefore I removed one of these objects.